src/Adapter/Customer/Group/Repository/GroupRepository.php

```diff
@@ -29,16 +29,51 @@
 
 use Group as CustomerGroup;
 use PrestaShop\Decimal\DecimalNumber;
+use PrestaShop\PrestaShop\Adapter\CoreException;
 use PrestaShop\PrestaShop\Core\Domain\Customer\Group\Exception\CannotAddGroupException;
 use PrestaShop\PrestaShop\Core\Domain\Customer\Group\Exception\GroupNotFoundException;
 use PrestaShop\PrestaShop\Core\Domain\Customer\Group\ValueObject\GroupId;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopId;
 use PrestaShop\PrestaShop\Core\Repository\AbstractObjectModelRepository;
+use PrestaShopException;
 
 /**
  * Provides methods to access Group data storage
  */
 class GroupRepository extends AbstractObjectModelRepository
 {
+    /**
+     * @param GroupId $customerGroupId
+     *
+     * @throws CoreException
+     * @throws GroupNotFoundException
+     *
+     * @return CustomerGroup
+     */
+    public function get(GroupId $customerGroupId): CustomerGroup
+    {
+        try {
+            $customerGroup = new CustomerGroup($customerGroupId->getValue());
```

Comment:
you should use already existing getObjectModel method from abstract class instead of doing the same manually.

Reply:
Fixed thank you